### ISSUE TITLE
Add basic wysiwyg options

### ISSUE THREAD

**Issue by oxhey** [ReDEnergy/Browsercast#10]
It would be nice to have something like the aloha editor come up when you click on a slide to edit it. That way you'd have basic controls such as bold /underline and text size and font.

Also a button to insert images would be nice.

**Comment by ReDEnergy** [comment]
Thanks a lot for the tip. I didn't know about aloha editor and seems to be quite ok for the project. I will analyze to see if I can add support for it as well a easy way to insert images.

**Comment by oxhey** [comment]
Thanks

**Comment by oxhey** [comment]
Do you know how long it will take to get a working version with these features ?

**Comment by ReDEnergy** [comment]
Sorry but it really depends on how easy I can get aloha editor to work with the existing code. Right now I don't know how to easy is to integrated it, I will take a look tomorrow (it's 1:00 AM over here) and if everything goes ok I might finish it the same day. Either way I will get back with more info (or the patch) tomorrow.

**Comment by ReDEnergy** [comment]
Sadly Aloha Editor is not a good solution. It behaves really strange with Reveal and makes the app unusable. Also, it seems to have quite some bugs ([Aloha issues](https://github.com/alohaeditor/Aloha-Editor/)) but let's say it's normal for an alpha state. I've search for similar libraries but sadly I haven't been able to find a good solution for this case. Tried to integrate [Raptor Editor](https://www.raptor-editor.com/) but it also proved to be incompatible with Browsercast... The problems is that Browsercast tries to offer an HTML5 editor by integrating with Reveal... Applying another library that does WSYIWYG seems to have some complex implications (events not working anymore, various bugs, etc), well, clearly more problems appear than what it solves. It's not really a coding problem but more like finding the best workarounds for each new library that I have to use... 
I will have to postpone this feature until I can find something more reliable that doesn't try to mess up with the HTML code or I will have enough time to build it myself. Sorry for the inconvenience.

**Comment by oxhey** [comment]
Have you looked into TinyMCE?

**Comment by ReDEnergy** [comment]
Probably I checked around 10 solution... some of them are like TinyMCE or CLEditor and not great because they mess up with the styling and HTML format which makes them really hard to implement - have in mind that the code must be left clean because It needs to be exported as clean HTML text and not with *hundreds* of attributes or new nested elements created. That makes it impossible for the exported to understand what must be kept and what not... I have to trust the library that doesn't mess with the code. 
Didn't noticed bootstrap-wysihtml5... I might give it a try. Thanks

**Comment by oxhey** [comment]
I think i may be able to get a editor to work . Could you tell me where i can find the code that changes the class of the slide from past/future to present . I need to add a new class to it.

**Comment by ReDEnergy** [comment]
Reveal.js handles that... Not sure how safe is to add classes because Reveal.js might change them every single time you change the slide. Probably you should try to use an attribute. One way to get the current slide if that's what you need is to use Reveal.getCurrentSlide() which gives you the current DOM section of the slide.

**Comment by oxhey** [comment]
Im trying to add editor to the class so i can get mediuam editor to work - https://github.com/daviferreira/medium-editor

**Comment by oxhey** [comment]
Iv found that instead of adding the class to the slides , if you add editor to the <div class="reveal"></div> in the index , you can use the editor on the slides . The only problem is that the background is now editable .

**Comment by ReDEnergy** [comment]
Mmmm.. I remember trying to add support for **medium** but not really sure. Yep.. like I said a lot of problems might appear.

**Comment by oxhey** [comment]
Yh , so it doesn't actually work . when you go to the next slides it completely messes up. I really want a wysisyg for this as its the only decent reveal editor

**Comment by ReDEnergy** [comment]
Yesterday I almost had Raptor Editor working but in the end proved to be really unstable.

**Comment by oxhey** [comment]
what made it unstable

**Comment by ReDEnergy** [comment]
Yes I know that an wysiwyg is a must and I'm really sorry for the inconvenience. Not happy about it but do you know about [slides.com](http://www.slides.com/) ? If you don't need all the features of Browsercast you can edit slides over there. Then if you need something from Browsercast you can import the presentation into Browsercast and continue to edit it. It might be what you need for know till I have the time to fix this feature.

**Comment by ReDEnergy** [comment]
Related to Raptor Editor... editing works for a while but then all of the sudden it doesn't trigger any new updates and even through JavaScript I can't get the current slide innerHTML in order to save it into the presentation. It's a really really strange bug and surely not easy to track having in mind that Raptor is a quite huge library.
